### ISSUE TITLE
Consider standardizing URL pointing into archives

### ISSUE THREAD

**Issue by mih** [datalad/datalad#3390]
A candidate would be the syntax proposed by Apache Commons VFS http://commons.apache.org/proper/commons-vfs/

They propose something along the lines of 

```
archtype:// arch-file-uri[! absolute-path]
```
leading to
```
jar:../lib/classes.jar!/META-INF/manifest.mf
zip:http://somehost/downloads/somefile.zip
jar:zip:outer.zip!/nested.jar!/somedir
jar:zip:outer.zip!/nested.jar!/some%21dir
tar:gz:http://anyhost/dir/mytar.tar.gz!/mytar.tar!/path/in/tar/README.txt
tgz:file://anyhost/dir/mytar.tgz!/somepath/somefile
```

More information: http://commons.apache.org/proper/commons-vfs/filesystems.html

**Comment by yarikoptic** [comment]
nice

immediate "concerns" for our adoption, and here is an example of our url `dl+archive:MD5E-s208546744--308754dd3551aeb99d7ecefeb6ec7f56.tar.gz#path=AllRats-Variables/AllRats-EVcontribs.mat&size=11255627`
- demands annotating with the type of the archive.  Even beyond pure storage type but with semantic - it is not just a `zip` but a `jar`.  ATM we rely on patool extracting so do not need to hardcode the type of archive
- we include `size` information for the file in the url.  I would need to check back on why I have added it. Most likely just a convenience but worth checking if could be dropped
Altogether, although I am all for using standards, not yet sure if it would provide **us** any benefits. In the long run though, with annex url metadata available via metadata , it might come handy for others of cause ;)

**Comment by mih** [comment]
Not that long of a run. I stumbled upon it, because I already have to parse this for the extractor released with -revolution 0.10.0

**Comment by mih** [comment]
No need for this issue to stay open.
